src/php/Build/Compile/FileCompiler.php

```diff
@@ -30,7 +30,9 @@ public function compileFile(string $src, string $dest, bool $enableSourceMaps):
             ->setSource($src)
             ->setIsEnabledSourceMaps($enableSourceMaps);
 
+        $GLOBALS['__phel']['phel\core']['*compile-mode*'] = true;
         $result = $this->compilerFacade->compile($phelCode, $options);
+        $GLOBALS['__phel']['phel\core']['*compile-mode*'] = false;
```

Comment:
I am a bit afraid of this `$GLOBALS` usage... I understand this is how we do it internally, but I think we could start thinking about encapsulating this usage somehow somewhere to have full control of this global variable and not usage it in different places.
What about creating a class on Phel's domain which will deal with this global PHP variable? this way it will be more explicit who usages this and how

Reply:
Yes definitely, `$GLOBALS` should be removed in future.